web/src/components/apps/Dashboard.jsx

```diff
@@ -180,17 +185,20 @@ class Dashboard extends Component {
       // the promise is resolved
 
       // TODO: use react-query to refetch this instead of the custom repeater
-      if (!this.props.app) {
+      if (!this.props.app || this.state.clusterId === null) {
         return;
       }
 
-      if (this.props.cluster?.id == "" && this.props.isHelmManaged === true) {
+      if (this.state.clusterId === "" && this.props.isHelmManaged === true) {
         // TODO: use a callback to update the state in the parent component
-        this.props.cluster.id = 0;
```

Comment:
The dashboard is hanging sometimes locally and in the tests and I think it might be this.